rfcs/0000-plan-dynanism.md

```diff
@@ -0,0 +1,251 @@
+---
+feature: plan-dynamism
+start-date: 2019-02-01
+author: John Ericson (@Ericson2314)
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+We need build plan dynamism -- interleaved building and planning -- to cope with the ever-growing world of language-specific package managers.
+Propose to allow derivations to build derivations, and depend on those built derivations, as the core primitive for this.
+Additionally, introduce a new primop to leverage this in making IFD, still the gold standard for ease of use, more efficient and compatible with `hydra.nixos.org`'s queue runner.
```

Comment:
Could you give an example what this is the gold standard for with ease of use? Usually IFD is just a mess to debug.. not sure I would call that the gold standard.

Reply:
I guess I don't mean to say it's great in relative terms, but the other options like `nix-build ` inside derivations recursive nix are far worse. I'm not really sure how to give good examples of these comparisons within largely unexpored areas without spilling tons of ink...?